src/Command/AnonymizeUsersCommand.php

```diff
@@ -0,0 +1,165 @@
+                'id' => [
+                    'help' => 'User id',
+                ],
+                'preserve' => [
+                    'help' => 'Users to preserve by id',
+                    'default' => '1',
+                ],
+            ]);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Anonymize Users command.
+     *
+     * $ bin/cake anonymize_users --help
+     *
+     * Usage:
+     * cake anonymize_users [options]
+     *
+     * Options:
+     *
+     * --id           User id
+     * --preserve     Users to preserve by id
+     * --help, -h     Display this help.
+     * --verbose, -v  Enable verbose output.
+     *
+     * # basic
+     * $ bin/cake anonymize_users --id 2
+     * $ bin/cake anonymize_users --preserve 1,2,3
+     */
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        $io->success('Start.');
+        LoggedUser::setUserAdmin();
+        /** @var \BEdita\Core\Model\Table\UsersTable $table */
+        $table = $this->fetchTable('Users');
+        $this->preserveUsers = array_map('intval', explode(',', $args->getOption('preserve')));
+        $query = $table->find()
+            ->where([
+                $table->aliasField('locked') => 0,
+                $table->aliasField('deleted') => 0,
+                $table->aliasField('id') . ' NOT IN' => $this->preserveUsers,
+            ]);
+        $id = $args->getOption('id');
+        if ($id) {
+            $query->where([$table->aliasField('id') => $id]);
+        }
+        $faker = Factory::create('it_IT');
+        $processed = $saved = $errors = 0;
+        /** @var \BEdita\Core\Model\Entity\User $user */
+        foreach ($this->objectsGenerator($query) as $user) {
+            $this->anonymize($faker, $user, $table, $io, $processed, $saved, $errors);
+        }
+        $io->out(sprintf('Users processed: %s', $processed));
+        $io->out(sprintf('Users saved: %s', $saved));
+        $io->out(sprintf('Users not saved: %s', $errors));
+        $io->success('Done.');
+
+        return null;
+    }
+
+    /**
+     * Update user.
+     *
+     * @param \Faker\Generator $faker Faker generator
+     * @param \BEdita\Core\Model\Entity\User $user User entity
+     * @param \BEdita\Core\Model\Table\UsersTable $table Users table
+     * @param \Cake\Console\ConsoleIo $io Console IO
+     * @param int $processed Processed users
+     * @param int $saved Saved users
+     * @param int $errors Errors
+     * @return void
+     */
+    public function anonymize(Generator $faker, User $user, UsersTable $table, ConsoleIo $io, int &$processed, int &$saved, int &$errors): void
+    {
+        $io->verbose(sprintf('Processing user %s [username: %s, email: %s]', $user->id, $user->username, $user->email));
+        $user->name = $faker->firstName();
+        $user->surname = $faker->lastName();
+        do {
+            $email = $faker->email();
+        } while ($table->exists([$table->aliasField('email') => $email]));
```

Comment:
In order to avoid this loop, since we're not really interested in those email addresses to be "realistic" whatsoever, we might build the email address ourselves using the object ID or its uname as the email username part:

```php
$user->email = sprintf('user-%d@%s', $user->id, $faker->safeEmailDomain());
// or:
$user->email = sprintf('%s@%s', $user->uname, $faker->safeEmailDomain());
```

A query is issued on every loop, and theoretically should be wrapped in the same transaction as the save in order for this check to be meaningful.